pontoon/base/tests/test_middleware.py

```diff
@@ -29,3 +32,69 @@ def test_EmailConsentMiddleware(client, member, settings):
     profile.save()
     response = member.client.get("/")
     assert response.status_code == 200
+
+
+class ThrottleIpMiddlewareTest(TestCase):
+    def setUp(self):
+        self.client = Client()
+
+    @override_settings(THROTTLE_MAX_COUNT=5)
+    @override_settings(THROTTLE_BLOCK_DURATION=60)
+    def test_throttle_limit_reached(self):
+        """Test that requests are throttled after the limit is reached."""
+        url = "/"
+
+        # Make 5 requests within the limit
+        for _ in range(5):
+            response = self.client.get(url, REMOTE_ADDR="192.168.0.1")
+            self.assertEqual(response.status_code, 200)
+
+        # 6th request should be throttled
+        response = self.client.get(url, REMOTE_ADDR="192.168.0.1")
+        self.assertEqual(response.status_code, 429)
+        self.assertEqual(response.json(), {"detail": "Too Many Requests"})
+
+        # Check that the IP remains blocked for the block duration
+        response = self.client.get(url, REMOTE_ADDR="192.168.0.1")
+        self.assertEqual(response.status_code, 429)
+
+    @override_settings(THROTTLE_MAX_COUNT=5)
+    @override_settings(THROTTLE_BLOCK_DURATION=2)
+    def test_throttle_reset_after_block_duration(self):
+        """Test that requests are allowed after the block duration expires."""
+        url = "/"
+
+        # Make 5 requests within the limit
+        for _ in range(5):
+            response = self.client.get(url, REMOTE_ADDR="192.168.0.2")
+            self.assertEqual(response.status_code, 200)
+
+        # 6th request should be throttled
+        response = self.client.get(url, REMOTE_ADDR="192.168.0.2")
+        self.assertEqual(response.status_code, 429)
+
+        # Wait for block duration to pass
+        time.sleep(3)
+
+        # Make another request after block duration
+        response = self.client.get(url, REMOTE_ADDR="192.168.0.2")
+        self.assertEqual(response.status_code, 200)
```

Comment:
This fails because `blocked_ip_192.168.0.2` was cleared out, but `throttle_ip_192.168.0.2` is still there (`THROTTLE_PERIOD` is the default setting, so 60s).